---
title: "circos_from_dataframe"
output: html_document
---


```{r}
# devtools::install_github("jokergoo/circlize")
library(circlize)
library(tidyverse)
```





```{r}
ptt_genome <- read_delim('NC_003030_edit.ptt', delim = '\t') 
ptt_plasmid <- read_delim('NC_001988_edit.ptt', delim = '\t') 
```



```{r}
# ptt <- bind_rows(ptt_genome, ptt_plasmid) 

ptt <- bind_rows(ptt_genome=ptt_genome, ptt_plasmid=ptt_plasmid, .id = 'dna')
ptt$dna <- str_replace_all(ptt$dna, "ptt_genome", "Chromosome")
ptt$dna <- str_replace_all(ptt$dna, "ptt_plasmid", "Plasmid")

# ptt %>% 
#   tail(100)

ptt <- ptt %>% 
  separate(Location, c("start", "end")) %>% 
  rename(name = Synonym)

ptt <- ptt %>% 
  select(start, end, name, dna) %>% 
  mutate(
    start = as.numeric(start), 
    end = as.numeric(end)
  )
```



```{r}
ptt
```

```{r}
(ann <- cbind(value = 0, ptt) %>% 
  select(dna, start, end, value, name))

# ann <- ann[sample(nrow(ann), 200), ]

(ann <- read_csv('core_metabolism.csv') %>% 
  rename(name = locus) %>% 
  inner_join(ann, ., by='name'))

```





```{r}
(glu_a <- read_csv('all.csv') %>% 
  group_by(accession, time, sugar) %>%
  summarise(
    mean = mean(RPKM)
    ) %>% 
  rename(name = accession) %>% 
  filter(time == 'A') %>% 
  filter(sugar == 'glu') %>% 
  left_join(., ptt, by='name') %>%  ###### join with ptt
  ungroup() %>% 
  select(dna, start, end, mean) %>% 
  rename(chr = 'dna') %>% 
  mutate(
    chr = as.factor(chr)
    )
)

(glu_b <- read_csv('all.csv') %>% 
  group_by(accession, time, sugar) %>%
  summarise(
    mean = mean(RPKM)
    ) %>% 
  rename(name = accession) %>% 
  filter(time == 'B') %>% 
  filter(sugar == 'glu') %>% 
  left_join(., ptt, by='name') %>%  ###### join with ptt
  ungroup() %>% 
  select(dna, start, end, mean) %>% 
  rename(chr = 'dna') %>% 
  mutate(
    chr = as.factor(chr)
    )
)

(glu_c <- read_csv('all.csv') %>% 
  group_by(accession, time, sugar) %>%
  summarise(
    mean = mean(RPKM)
    ) %>% 
  rename(name = accession) %>% 
  filter(time == 'C') %>% 
  filter(sugar == 'glu') %>% 
  left_join(., ptt, by='name') %>%  ###### join with ptt
  ungroup() %>% 
  select(dna, start, end, mean) %>% 
  rename(chr = 'dna') %>% 
  mutate(
    chr = as.factor(chr)
    )
)

summary(glu_a)

glu_a$mean <- scales::squish(glu_a$mean, quantile(glu_a$mean, c(0.025, 0.975)))
glu_b$mean <- scales::squish(glu_b$mean, quantile(glu_b$mean, c(0.025, 0.975)))
glu_c$mean <- scales::squish(glu_c$mean, quantile(glu_c$mean, c(0.025, 0.975)))

summary(glu_a)
```




















# Example

http://www.roymfrancis.com/beautiful-circos-plots-in-r/



```{r}
ref <- data.frame(V1=c("Chromosome","Plasmid"))

ref$V2 <- c(3940781, 191953)

ref
```


```{r}
RColorBrewer::brewer.pal(3, "Dark2")
```


```{r}
circos.clear()
pdf("glucose.pdf", 5, 5)
                  

col_text <- "grey40"
circos.par("track.height"=0.8, gap.degree=5, cell.padding=c(0,0,0,0))
circos.initialize(factors=ref$V1, xlim=matrix(c(rep(0,2), ref$V2), ncol=2))

# genomes
circos.track(ylim=c(0,1),panel.fun=function(x,y) {
chr=CELL_META$sector.index
xlim=CELL_META$xlim
ylim=CELL_META$ylim
circos.text(mean(xlim),mean(ylim),chr,cex=0.5,col=col_text,
facing="bending.inside",niceFacing=TRUE)
},bg.col="grey90",bg.border=F,track.height=0.06)


# genomes x axis
brk <- c(0,0.5,1,1.5,2,2.5,3,3.5,4,4.5,5)*10^6
circos.track(track.index = get.current.track.index(), panel.fun = function(x, y) {
  circos.axis(h="top",major.at=brk,labels=round(brk/10^6,1),labels.cex=0.4,
  col=col_text,labels.col=col_text,lwd=0.7,labels.facing="clockwise")
},bg.border=F)

# gc content
circos.track(factors=glu_a$chr, x=glu_a$start, y=glu_a$mean, panel.fun=function(x,y) {
  circos.lines(x,y,col="#1B9E77",lwd=0.6)
  circos.segments(x0=0,x1=max(ref$V2),y0=0.3,y1=0.3,lwd=0.6,lty="11",col="grey90")
  circos.segments(x0=0,x1=max(ref$V2),y0=0.5,y1=0.5,lwd=0.6,lty="11",col="grey90")
  circos.segments(x0=0,x1=max(ref$V2),y0=0.7,y1=0.7,lwd=0.6,lty="11",col="grey90")
},ylim=range(glu_a$mean),track.height=0.08,bg.border=F)
# gc y axis
circos.yaxis(at=c(0.3,0.5,0.7),labels.cex=0.25,lwd=0,tick.length=0,labels.col=col_text,col="#FFFFFF")


# gc content
circos.track(factors=glu_b$chr, x=glu_b$start, y=glu_b$mean, panel.fun=function(x,y) {
  circos.lines(x,y,col="#D95F02",lwd=0.6)
  circos.segments(x0=0,x1=max(ref$V2),y0=0.3,y1=0.3,lwd=0.6,lty="11",col="grey90")
  circos.segments(x0=0,x1=max(ref$V2),y0=0.5,y1=0.5,lwd=0.6,lty="11",col="grey90")
  circos.segments(x0=0,x1=max(ref$V2),y0=0.7,y1=0.7,lwd=0.6,lty="11",col="grey90")
},ylim=range(glu_b$mean),track.height=0.08,bg.border=F)
# gc y axis
circos.yaxis(at=c(0.3,0.5,0.7),labels.cex=0.25,lwd=0,tick.length=0,labels.col=col_text,col="#FFFFFF")

# gc content
circos.track(factors=glu_c$chr, x=glu_c$start, y=glu_c$mean, panel.fun=function(x,y) {
  circos.lines(x,y,col="#7570B3",lwd=0.6)
  circos.segments(x0=0,x1=max(ref$V2),y0=0.3,y1=0.3,lwd=0.6,lty="11",col="grey90")
  circos.segments(x0=0,x1=max(ref$V2),y0=0.5,y1=0.5,lwd=0.6,lty="11",col="grey90")
  circos.segments(x0=0,x1=max(ref$V2),y0=0.7,y1=0.7,lwd=0.6,lty="11",col="grey90")
},ylim=range(glu_c$mean),track.height=0.08,bg.border=F)
# gc y axis
circos.yaxis(at=c(0.3,0.5,0.7),labels.cex=0.25,lwd=0,tick.length=0,labels.col=col_text,col="#FFFFFF")


# gene labels
circos.genomicLabels(ann,labels.column=5,cex=0.25,col=col_text,line_lwd=0.5,line_col="grey80",
side="inside",connection_height=0.05,labels_height=0.04)
```


















# As a function



## RNAseq


```{r}

circos_fun <- function(sugar_name){

  (glu_a <- read_csv('all.csv') %>%  
    group_by(accession, time, sugar) %>%
    summarise(
      mean = mean(RPKM)
      ) %>% 
    rename(name = accession) %>% 
    filter(time == 'A') %>% 
    filter(sugar == sugar_name) %>% 
    left_join(., ptt, by='name') %>%  ###### join with ptt
    ungroup() %>% 
    select(dna, start, end, mean) %>% 
    rename(chr = 'dna') %>% 
    mutate(
      chr = as.factor(chr)
      )
  )
  
  (glu_b <- read_csv('all.csv') %>% 
    group_by(accession, time, sugar) %>%
    summarise(
      mean = mean(RPKM)
      ) %>% 
    rename(name = accession) %>% 
    filter(time == 'B') %>% 
    filter(sugar == sugar_name) %>% 
    left_join(., ptt, by='name') %>%  ###### join with ptt
    ungroup() %>% 
    select(dna, start, end, mean) %>% 
    rename(chr = 'dna') %>% 
    mutate(
      chr = as.factor(chr)
      )
  )
  
  (glu_c <- read_csv('all.csv') %>% 
    group_by(accession, time, sugar) %>%
    summarise(
      mean = mean(RPKM)
      ) %>% 
    rename(name = accession) %>% 
    filter(time == 'C') %>% 
    filter(sugar == sugar_name) %>% 
    left_join(., ptt, by='name') %>%  ###### join with ptt
    ungroup() %>% 
    select(dna, start, end, mean) %>% 
    rename(chr = 'dna') %>% 
    mutate(
      chr = as.factor(chr)
      )
  )
  
  
  glu_a$mean <- scales::squish(glu_a$mean, quantile(glu_a$mean, c(0.025, 0.975)))
  glu_b$mean <- scales::squish(glu_b$mean, quantile(glu_b$mean, c(0.025, 0.975)))
  glu_c$mean <- scales::squish(glu_c$mean, quantile(glu_c$mean, c(0.025, 0.975)))
  
  
  ref <- data.frame(V1=c("Chromosome","Plasmid"))
  
  ref$V2 <- c(3940781, 191953)
  
  
  circos.clear()
  pdf(paste(sugar_name, ".pdf", sep = ""), 5, 5)
                    
  
  col_text <- "grey40"
  circos.par("track.height"=0.8, gap.degree=5, cell.padding=c(0,0,0,0))
  circos.initialize(factors=ref$V1, xlim=matrix(c(rep(0,2), ref$V2), ncol=2))
  
  # genomes
  circos.track(ylim=c(0,1),panel.fun=function(x,y) {
  chr=CELL_META$sector.index
  xlim=CELL_META$xlim
  ylim=CELL_META$ylim
  circos.text(mean(xlim),mean(ylim),chr,cex=0.5,col=col_text,
  facing="bending.inside",niceFacing=TRUE)
  },bg.col="grey90",bg.border=F,track.height=0.06)
  
  
  # genomes x axis
  brk <- c(0,0.5,1,1.5,2,2.5,3,3.5,4,4.5,5)*10^6
  circos.track(track.index = get.current.track.index(), panel.fun = function(x, y) {
    circos.axis(h="top",major.at=brk,labels=round(brk/10^6,1),labels.cex=0.4,
    col=col_text,labels.col=col_text,lwd=0.7,labels.facing="clockwise")
  },bg.border=F)
  
  # gc content
  circos.track(factors=glu_a$chr, x=glu_a$start, y=glu_a$mean, panel.fun=function(x,y) {
    circos.lines(x,y,col="#1B9E77",lwd=0.6)
    circos.segments(x0=0,x1=max(ref$V2),y0=0.3,y1=0.3,lwd=0.6,lty="11",col="grey90")
    circos.segments(x0=0,x1=max(ref$V2),y0=0.5,y1=0.5,lwd=0.6,lty="11",col="grey90")
    circos.segments(x0=0,x1=max(ref$V2),y0=0.7,y1=0.7,lwd=0.6,lty="11",col="grey90")
  },ylim=range(glu_a$mean),track.height=0.08,bg.border=F)
  # gc y axis
  circos.yaxis(at=c(0.3,0.5,0.7),labels.cex=0.25,lwd=0,tick.length=0,labels.col=col_text,col="#FFFFFF")
  
  
  # gc content
  circos.track(factors=glu_b$chr, x=glu_b$start, y=glu_b$mean, panel.fun=function(x,y) {
    circos.lines(x,y,col="#D95F02",lwd=0.6)
    circos.segments(x0=0,x1=max(ref$V2),y0=0.3,y1=0.3,lwd=0.6,lty="11",col="grey90")
    circos.segments(x0=0,x1=max(ref$V2),y0=0.5,y1=0.5,lwd=0.6,lty="11",col="grey90")
    circos.segments(x0=0,x1=max(ref$V2),y0=0.7,y1=0.7,lwd=0.6,lty="11",col="grey90")
  },ylim=range(glu_b$mean),track.height=0.08,bg.border=F)
  # gc y axis
  circos.yaxis(at=c(0.3,0.5,0.7),labels.cex=0.25,lwd=0,tick.length=0,labels.col=col_text,col="#FFFFFF")
  
  # gc content
  circos.track(factors=glu_c$chr, x=glu_c$start, y=glu_c$mean, panel.fun=function(x,y) {
    circos.lines(x,y,col="#7570B3",lwd=0.6)
    circos.segments(x0=0,x1=max(ref$V2),y0=0.3,y1=0.3,lwd=0.6,lty="11",col="grey90")
    circos.segments(x0=0,x1=max(ref$V2),y0=0.5,y1=0.5,lwd=0.6,lty="11",col="grey90")
    circos.segments(x0=0,x1=max(ref$V2),y0=0.7,y1=0.7,lwd=0.6,lty="11",col="grey90")
  },ylim=range(glu_c$mean),track.height=0.08,bg.border=F)
  # gc y axis
  circos.yaxis(at=c(0.3,0.5,0.7),labels.cex=0.25,lwd=0,tick.length=0,labels.col=col_text,col="#FFFFFF")
  
  
  # gene labels
  circos.genomicLabels(ann,labels.column=5,cex=0.25,col=col_text,line_lwd=0.5,line_col="grey80",
  side="inside",connection_height=0.05,labels_height=0.04)
}

# glu				
# gal_acid				
# gluconate				
# man

circos_fun('man')
```



## Proteomics

```{r}

circos_fun <- function(sugar_name){

  (glu_a <- read_csv('all_proteomics.csv') %>%  
    group_by(accession, time, sugar) %>%
    summarise(
      mean = mean(spectral_count)
      ) %>% 
    rename(name = accession) %>% 
    filter(time == 'A') %>% 
    filter(sugar == sugar_name) %>% 
    left_join(., ptt, by='name') %>%  ###### join with ptt
    ungroup() %>% 
    select(dna, start, end, mean) %>% 
    rename(chr = 'dna') %>% 
    mutate(
      chr = as.factor(chr)
      )
  )
  
  (glu_b <- read_csv('all_proteomics.csv') %>% 
    group_by(accession, time, sugar) %>%
    summarise(
      mean = mean(spectral_count)
      ) %>% 
    rename(name = accession) %>% 
    filter(time == 'B') %>% 
    filter(sugar == sugar_name) %>% 
    left_join(., ptt, by='name') %>%  ###### join with ptt
    ungroup() %>% 
    select(dna, start, end, mean) %>% 
    rename(chr = 'dna') %>% 
    mutate(
      chr = as.factor(chr)
      )
  )
  
  (glu_c <- read_csv('all_proteomics.csv') %>% 
    group_by(accession, time, sugar) %>%
    summarise(
      mean = mean(spectral_count)
      ) %>% 
    rename(name = accession) %>% 
    filter(time == 'C') %>% 
    filter(sugar == sugar_name) %>% 
    left_join(., ptt, by='name') %>%  ###### join with ptt
    ungroup() %>% 
    select(dna, start, end, mean) %>% 
    rename(chr = 'dna') %>% 
    mutate(
      chr = as.factor(chr)
      )
  )
  
  
  glu_a$mean <- scales::squish(glu_a$mean, quantile(glu_a$mean, c(0.025, 0.975)))
  glu_b$mean <- scales::squish(glu_b$mean, quantile(glu_b$mean, c(0.025, 0.975)))
  glu_c$mean <- scales::squish(glu_c$mean, quantile(glu_c$mean, c(0.025, 0.975)))
  
  
  ref <- data.frame(V1=c("Chromosome","Plasmid"))
  
  ref$V2 <- c(3940781, 191953)
  
  
  circos.clear()
  pdf(paste(sugar_name, "_proteomics.pdf", sep = ""), 5, 5)
                    
  
  col_text <- "grey40"
  circos.par("track.height"=0.8, gap.degree=5, cell.padding=c(0,0,0,0))
  circos.initialize(factors=ref$V1, xlim=matrix(c(rep(0,2), ref$V2), ncol=2))
  
  # genomes
  circos.track(ylim=c(0,1),panel.fun=function(x,y) {
  chr=CELL_META$sector.index
  xlim=CELL_META$xlim
  ylim=CELL_META$ylim
  circos.text(mean(xlim),mean(ylim),chr,cex=0.5,col=col_text,
  facing="bending.inside",niceFacing=TRUE)
  },bg.col="grey90",bg.border=F,track.height=0.06)
  
  
  # genomes x axis
  brk <- c(0,0.5,1,1.5,2,2.5,3,3.5,4,4.5,5)*10^6
  circos.track(track.index = get.current.track.index(), panel.fun = function(x, y) {
    circos.axis(h="top",major.at=brk,labels=round(brk/10^6,1),labels.cex=0.4,
    col=col_text,labels.col=col_text,lwd=0.7,labels.facing="clockwise")
  },bg.border=F)
  
  # gc content
  circos.track(factors=glu_a$chr, x=glu_a$start, y=glu_a$mean, panel.fun=function(x,y) {
    circos.lines(x,y,col="#1B9E77",lwd=0.6)
    circos.segments(x0=0,x1=max(ref$V2),y0=0.3,y1=0.3,lwd=0.6,lty="11",col="grey90")
    circos.segments(x0=0,x1=max(ref$V2),y0=0.5,y1=0.5,lwd=0.6,lty="11",col="grey90")
    circos.segments(x0=0,x1=max(ref$V2),y0=0.7,y1=0.7,lwd=0.6,lty="11",col="grey90")
  },ylim=range(glu_a$mean),track.height=0.08,bg.border=F)
  # gc y axis
  circos.yaxis(at=c(0.3,0.5,0.7),labels.cex=0.25,lwd=0,tick.length=0,labels.col=col_text,col="#FFFFFF")
  
  
  # gc content
  circos.track(factors=glu_b$chr, x=glu_b$start, y=glu_b$mean, panel.fun=function(x,y) {
    circos.lines(x,y,col="#D95F02",lwd=0.6)
    circos.segments(x0=0,x1=max(ref$V2),y0=0.3,y1=0.3,lwd=0.6,lty="11",col="grey90")
    circos.segments(x0=0,x1=max(ref$V2),y0=0.5,y1=0.5,lwd=0.6,lty="11",col="grey90")
    circos.segments(x0=0,x1=max(ref$V2),y0=0.7,y1=0.7,lwd=0.6,lty="11",col="grey90")
  },ylim=range(glu_b$mean),track.height=0.08,bg.border=F)
  # gc y axis
  circos.yaxis(at=c(0.3,0.5,0.7),labels.cex=0.25,lwd=0,tick.length=0,labels.col=col_text,col="#FFFFFF")
  
  # gc content
  circos.track(factors=glu_c$chr, x=glu_c$start, y=glu_c$mean, panel.fun=function(x,y) {
    circos.lines(x,y,col="#7570B3",lwd=0.6)
    circos.segments(x0=0,x1=max(ref$V2),y0=0.3,y1=0.3,lwd=0.6,lty="11",col="grey90")
    circos.segments(x0=0,x1=max(ref$V2),y0=0.5,y1=0.5,lwd=0.6,lty="11",col="grey90")
    circos.segments(x0=0,x1=max(ref$V2),y0=0.7,y1=0.7,lwd=0.6,lty="11",col="grey90")
  },ylim=range(glu_c$mean),track.height=0.08,bg.border=F)
  # gc y axis
  circos.yaxis(at=c(0.3,0.5,0.7),labels.cex=0.25,lwd=0,tick.length=0,labels.col=col_text,col="#FFFFFF")
  
  
  # gene labels
  circos.genomicLabels(ann,labels.column=5,cex=0.25,col=col_text,line_lwd=0.5,line_col="grey80",
  side="inside",connection_height=0.05,labels_height=0.04)
}

circos_fun('man')
```


```{r}
read_csv('~/Desktop/all_proteomics.csv') %>% 
  select(sugar) %>% 
  unique
```



